R/pivot-wide.R

```diff
@@ -276,11 +297,22 @@ pivot_wider_spec <- function(data,
   }
   values_fill <- values_fill[intersect(names(values_fill), values_from_cols)]
 
-  # Figure out rows in output.
+  if (!is_bool(id_expand)) {
+    abort("`id_expand` must be a single `TRUE` or `FALSE`.")
+  }
+
   # Early conversion to tibble because data.table returns zero rows if
-  # zero cols are selected.
-  rows <- as_tibble(data)
-  rows <- rows[id_cols]
+  # zero cols are selected. Also want to avoid the grouped-df behavior
+  # of `complete()`.
+  data <- as_tibble(data)
+  data <- data[vec_unique(c(id_cols, names_from_cols, values_from_cols))]
+
+  if (id_expand) {
+    data <- complete(data, !!!syms(id_cols), fill = values_fill, explicit = FALSE)
```

Comment:
Out of interest, why `!!!syms(id_cols)` and not `any_of(id_cols)`?

Reply:
`complete()` and `expand()` use mutate like semantics, not tidyselect

Reply:
Oh duh